hail/python/hail/methods/impex.py

```diff
@@ -1293,12 +1293,68 @@ def import_gen(path,
     -------
     :class:`.MatrixTable`
     """
-    path = wrap_to_list(path)
+    
+    gen_table = import_lines(path, min_partitions)
+    sample_table = import_lines(sample_file)
     rg = reference_genome.name if reference_genome else None
+    contig_recoding = contig_recoding
     if contig_recoding is None:
-        contig_recoding = {}
-    return MatrixTable(ir.MatrixRead(ir.MatrixGENReader(
-        path, sample_file, chromosome, min_partitions, tolerance, rg, contig_recoding, skip_invalid_loci)))
+        contig_recoding = hl.empty_dict(hl.tstr, hl.tstr)
+    else:
+        contig_recoding = hl.dict(contig_recoding)
+
+    gen_table = gen_table.transmute(data=gen_table.text.split(' '))
+
+    if chromosome is None:
+        index = 5
```

Comment:
index on its own is a slightly confusing name for a constant value. What is it the index of?